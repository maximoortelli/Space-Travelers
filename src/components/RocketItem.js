import { React } from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/rocketItem.module.css';
import RocketBadge from './RocketBadge';

function RocketItem({
  rocketId,
  rocketName,
  rocketDescription,
  flickrImages,
  isReserved,
  onClick,
}) {
  return (
    <div className={styles.rocketItem}>
      <img src={flickrImages} alt={`${rocketName} rocket`} className={styles.rocketImg} />
      <div className={styles.rocketContent}>
        <h3 className={styles.rocketName}>{rocketName}</h3>
        <p className={styles.rocketDescription}>
          {isReserved && <RocketBadge />}
          {rocketDescription}
        </p>
        {!isReserved && (
          <button className={styles.reserveBtn} type="button" onClick={() => onClick(rocketId)}>
            Reserve Rocket
          </button>
        )}
        {isReserved && (
          <button className={styles.cancelBtn} type="button" onClick={() => onClick(rocketId)}>
            Cancel Reservation
          </button>
        )}
      </div>
    </div>
  );
}

RocketItem.propTypes = {
  rocketId: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  rocketDescription: PropTypes.string.isRequired,
  flickrImages: PropTypes.string.isRequired,
  isReserved: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default RocketItem;
