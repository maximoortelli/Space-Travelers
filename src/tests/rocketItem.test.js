import renderer from 'react-test-renderer';
import React from 'react';
import RocketItem from '../components/RocketItem';

test('Check for correct item rendering', () => {
  const clickReserve = jest.fn();

  const component = renderer.create(
    <RocketItem
      rocketId="rocketIdTest"
      rocketName="rocket Name"
      rocketDescription="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla diam diam, ultricies quis maximus nec, maximus sit amet diam. Nam mollis faucibus eros, ut malesuada neque consequat in. Aenean in odio ut lectus varius dapibus quis et nulla. Suspendisse nibh augue, pharetra eget accumsan sed, sagittis vitae purus. In vehicula sem sit amet lorem consectetur, at aliquam lectus fermentum. Nunc volutpat vestibulum ante. Nulla justo est, malesuada et fermentum vitae, varius in felis. Phasellus dignissim justo a sapien iaculis, nec tempor dui congue."
      flickrImages="https://example.com/rocket.jpg"
      isReserved={false}
      onClick={clickReserve}
    />,
  );
  const treeRocketItem = component.toJSON();

  expect(treeRocketItem).toMatchSnapshot();
});
