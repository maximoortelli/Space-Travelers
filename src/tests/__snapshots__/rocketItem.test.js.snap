// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check for correct item rendering 1`] = `
<div
  className="rocketItem"
>
  <img
    alt="rocket Name rocket"
    className="rocketImg"
    src="https://example.com/rocket.jpg"
  />
  <div
    className="rocketContent"
  >
    <h3
      className="rocketName"
    >
      rocket Name
    </h3>
    <p
      className="rocketDescription"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla diam diam, ultricies quis maximus nec, maximus sit amet diam. Nam mollis faucibus eros, ut malesuada neque consequat in. Aenean in odio ut lectus varius dapibus quis et nulla. Suspendisse nibh augue, pharetra eget accumsan sed, sagittis vitae purus. In vehicula sem sit amet lorem consectetur, at aliquam lectus fermentum. Nunc volutpat vestibulum ante. Nulla justo est, malesuada et fermentum vitae, varius in felis. Phasellus dignissim justo a sapien iaculis, nec tempor dui congue.
    </p>
    <button
      className="reserveBtn"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
